internal/cmdrender/executor.go

```diff
@@ -138,6 +160,8 @@ type hydrationContext struct {
 
 	// imagePullPolicy controls the image pulling behavior.
 	imagePullPolicy fnruntime.ImagePullPolicy
+
+	disableCLIOutput bool
```

Comment:
document this field.

What exactly are you disabling here? In no scenario, we should suppress non-structured CLI output of eval or render. 

Also, there should be no changes to unstructured output unless it's documented and tested by asserting against expected `stdout` and `stderr` in the test harness.

Reply:
We only disable stdout and not stderr. Updated description.

Reply:
This flag is disabling calls to `pr.Printf`, which means you're losing the non-structured output, no? This also tells me we're missing some test coverage for this behavior. Both `eval` and `render` should send non-structured output to `stderr`. I've mentioned this in bunch of other PRs too. @droot @Shell32-Natsu

Reply:
So far, we were suppressing (`disableOutput` flag) non-structured output if output was being written to `stdout`. And the reasoning was that we weren't sure if writing non-structured output to `stderr` was correct thing to do even if the command didn't fail (zero exit code) as per unix guidelines. As per https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout, looks like we can print diagnostic info on `stderr` (fn outputs falls in this category IMO), so we can write the output to `stderr` always and get rid of `disableOutput` from the code completely. And then, strict checking of `stderr` can be enforced in the `e2e` tests.

@phanimarupaka if you want to do it in this PR (will remove bunch of code related to CLI output), it is fine, else can also be done in a follow up.

Reply:
Better to do in separate PR as eval also follows the same pattern and we can do entire cleanup there.